### ISSUE TITLE
FP16 profermance on GTX1080

### ISSUE THREAD

**Issue by 7oud** [tensorflow/tensorflow#25405]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Windows10
- TensorFlow installed from (source or binary):
Binary(PIP install)
- TensorFlow version (use command below):
1.12.0
- Python version:
3.6.2
- CUDA/cuDNN version:
CUDA 9.0 / cuDNN 7.3.1
- GPU model and memory:
GeForce GTX 1080, 8GB

**Describe the current behavior**
I try to evaluate the accuracy of fp16 precision on GTX 1080 GPU. As we known, GTX1080 doesn't have
tensor core to supporting FP16 calc, but my test reuslt shows that inference with fp16 is faster than fp32 by 20%-30%, could you give an explanation? 

Here is my operation step:
1. train a model using fp32 precision and save the checkpoint.
2. in order to inference with fp16, load the checkpoint and convert each param to fp16 type
```
    reader = tf.train.NewCheckpointReader(model_file)
    var_to_map = reader.get_variable_to_shape_map()

    val_f16 = {}
    for key, val in var_to_map.items():
        val_f16[key] = reader.get_tensor(key).astype(np.float16)
```
3. rewrite the network and pass the fp16 params to initialize each layer, then run the network.
```
    weight_name = scope_name + '/' + get_layer_str() + '/' + 'weight'
    initw = inits[weight_name]
    weight = tf.get_variable('weight', dtype=initw.dtype, initializer=initw)
    out = tf.nn.conv2d(self.get_output(), weight, strides=[1, stride, stride, 1], padding='SAME')
```

I got a approximate accuracy from fp16 reuslts, and its faster than fp32 by 20%-30%. In order to verify the speed of fp16, I construct a simple net with several dense layers, which are initialzed with fp16 random values, and run the forward procedure only. But fp16 is slower than fp32. ...

**Question**
1. Why FP16 is faster on a Pascal GTX GPU, what hardware units were used?
2. Is there any better way to inference with fp16 using a pretrained fp32 model?
3. If the TensorRT is the normal method to inference with fp16?
4. If using RTX 2080 with tensor core, we will get a dramatic performance in fp16?

**Comment by jvishnuvardhan** [comment]
I am closing the issue as it a support question. In future, please post this kind of support questions at Stackoverflow. Thanks!
